src/crate/client/sqlalchemy/dialect.py

```diff
@@ -39,26 +39,38 @@
 TYPES_MAP = {
     "boolean": sqltypes.Boolean,
     "short": sqltypes.SmallInteger,
+    "smallint": sqltypes.SmallInteger,
     "timestamp": sqltypes.TIMESTAMP,
+    "timestamp with time zone": sqltypes.TIMESTAMP,
     "object": Object,
     "integer": sqltypes.Integer,
     "long": sqltypes.NUMERIC,
+    "bigint": sqltypes.NUMERIC,
     "double": sqltypes.DECIMAL,
+    "double precision": sqltypes.DECIMAL,
     "object_array": ObjectArray,
     "float": sqltypes.Float,
-    "string": sqltypes.String
+    "real": sqltypes.Float,
+    "string": sqltypes.String,
+    "text": sqltypes.String
 }
 try:
     # SQLAlchemy >= 1.1
     from sqlalchemy.types import ARRAY
     TYPES_MAP["integer_array"] = ARRAY(sqltypes.Integer)
     TYPES_MAP["boolean_array"] = ARRAY(sqltypes.Boolean)
     TYPES_MAP["short_array"] = ARRAY(sqltypes.SmallInteger)
+    TYPES_MAP["smallint_array"] = ARRAY(sqltypes.SmallInteger)
     TYPES_MAP["timestamp_array"] = ARRAY(sqltypes.TIMESTAMP)
+    TYPES_MAP["timestamp with time zone_array"] = ARRAY(sqltypes.TIMESTAMP)
```

Comment:
That _array suffix looks strange when used with types that contain whitespaces.

Reply:
Yes, but that's what is returned by CrateDB